internal/app/logout.go

```diff
@@ -0,0 +1,188 @@
+/*
+Copyright 2020 The kconnect Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"errors"
+	"strings"
+
+	historyv1alpha "github.com/fidelity/kconnect/api/v1alpha1"
+	"github.com/fidelity/kconnect/pkg/k8s/kubeconfig"
+	"github.com/fidelity/kconnect/pkg/provider"
+	"go.uber.org/zap"
+	"gopkg.in/ini.v1"
+
+	"github.com/fidelity/kconnect/pkg/aws/awsconfig"
+)
+
+type LogoutParams struct {
+	CommonConfig
+	HistoryConfig
+	KubernetesConfig
+
+	All     bool
+	Alias   string
+	IDs     string
+	Context *provider.Context
+}
+
+func (a *App) Logout(params *LogoutParams) error {
+
+	entries, err := a.getClustersToLogout(params)
+	if err != nil {
+		return err
+	}
+	if entries == nil || len(entries.Items) == 0 {
+		return ErrNoEntriesFound
+	}
+	for i := range entries.Items {
+		err = a.doLogout(params, &entries.Items[i])
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (a *App) getClustersToLogout(params *LogoutParams) (*historyv1alpha.HistoryEntryList, error) {
+	entries := historyv1alpha.NewHistoryEntryList()
+	var err error
+
+	switch {
+	case params.All:
+		zap.S().Infof("will log out of all clusters")
+		entries, err = a.historyStore.GetAllSortedByLastUsed()
+		if err != nil {
+			return nil, err
+		}
+	case params.Alias != "":
+		aliasList := strings.Split(params.Alias, ",")
+		for _, alias := range aliasList {
+			entry, err := a.historyStore.GetByAlias(alias)
+			if err != nil {
+				return nil, err
+			}
+			if entry == nil {
+				return nil, ErrAliasNotFound
+			}
+			entries.Items = append(entries.Items, *entry)
+		}
+		fallthrough
+	case params.IDs != "":
+		idsList := strings.Split(params.IDs, ",")
+		for _, id := range idsList {
+			entry, err := a.historyStore.GetByID(id)
+			if err != nil {
+				return nil, err
+			}
+			if entry == nil {
+				return nil, ErrAliasNotFound
+			}
+			entries.Items = append(entries.Items, *entry)
+		}
+	default:
+		zap.S().Infof("Logging out of current cluster")
+		entry, err := a.historyStore.GetLastModified(0)
+		if err != nil {
+			return nil, err
+		}
+		entries.Items = append(entries.Items, *entry)
+	}
+	return entries, nil
+}
+
+func (a *App) doLogout(params *LogoutParams, entry *historyv1alpha.HistoryEntry) error {
+
+	// TODO use const for provider
+	switch entry.Spec.Provider {
```

Comment:
I wonder if in the future we should move the provider specific logout logic into the provider. But we can discuss that as party of a future provider interface refactor.

Reply:
Agreed, that would make sense